src/TraceEvent/Stacks/CallTree.cs

```diff
@@ -271,8 +272,17 @@ struct TreeCacheEntry
             public CallTreeNode Tree;
         }
 
-        private CallTreeNode FindTreeNode(StackSourceCallStackIndex stack)
+        private CallTreeNode FindTreeNode(StackSourceCallStackIndex stack, int depth = 0)
         {
+            if (depth > 400)
```

Comment:
Can you put a comment about why this is here and what the magic number 400 is (that you are restarting the recursive function on a new thread so you don't get stack overflow.  Give 400 a symbolicName like 'MaxCallStacksBeforeNeedingANewThread' (or something like that).